2022/FG-IR-22-375.json

```
@@ -40,43 +40,77 @@
             {
               "nodes": [
                 {
-                  "description": "FortiProxy:",
+                  "description": "FortiProxy: greater equal 7.2.0, less equal 7.2.3, fixed in [7.2.4]",
                   "cpe": "cpe:2.3:a:fortinet:fortiproxy:*:*:*:*:*:*:*:*",
```

e.g. https://github.com/vulsio/vuls-data-raw-fortinet/pull/3#pullrequestreview-2030624480
```suggestion
                  "cpe": "cpe:2.3:o:fortinet:fortiproxy:*:*:*:*:*:*:*:*",
```

---

I chose `a` because in nvd and cvedetails the cpes for fortiproxy are with `a` and not `o`
https://nvd.nist.gov/vuln/detail/CVE-2023-33305
https://www.cvedetails.com/cve/CVE-2023-33305/

---

As far as I read the advisory, it looks like it specifies the firmware version...
https://www.fortiguard.com/psirt/FG-IR-22-375

One of the purposes of this data source is to accurately represent things that are wrong in NVD, so it does not always have to be the same representation as NVD.

---

i think what they meant is that the `firmware upgrade function` in the FortiOS and FortiProxy prdocy is vulnerable and not the firmware itseld and you need to upgrade FortiOS/FortiPRoxy to fix the issue
an attacker can use `specially crafted firmware image` to exploit the `upgrade funtion` in the FortiOS, so I guess what is vulnerable is the "application" FortiProxy and not the HW or the OS. So I guess it should be `a` :-)

---

If updating the firmware version is required to fixed the firmware upgrade function, then the firmware upgrade function can be considered firmware, and it can be thought of as `o` rather than `a`.

There are also doubts about the description of NVD.
Why is it `o` only for FortiOS?
As you pointed out, I think `a` is sufficient for FortiOS when evaluating the firmware upgrade function alone.
https://nvd.nist.gov/vuln/detail/CVE-2023-33305

---

it is a little bit confusing... and i am got more confused now
what is your final suggestion?... for me it does not matter if it is an `a` or an `o`. so let's go with it
you want an `o` in this specific firmware cve?

---

Sorry for confusing you.
Personally, I think it should be `o`.